R/Data-class.R

```diff
@@ -30,147 +30,117 @@ NULL
+# Data-class ----
 
-                     o$check(all(object@y %in% c(0, 1)),
-                             "DLT vector y can only have 0 or 1 values")
-                     if(!object@placebo){
-                        o$check(all(tapply(X=object@x,
-                                           INDEX=object@cohort,
-                                           FUN=
-                                            function(doses){length(unique(doses))}) == 1),
-                             "there must be only one dose level per cohort")
-                     }else{
-                        o$check(all(tapply(X=object@x,
-                                          INDEX=object@cohort,
-                                          FUN=
-                                            function(doses){
-                                              doses <- doses[doses != object@doseGrid[1]]
-                                              length(unique(doses))}) == 1),
-                               "There must be only one dose level, other than placebo, per cohort. In addition a cohort with only placebo is not allowed")
-                     }
-
-                     o$check(all(object@x %~% object@doseGrid),
-                             "dose values in x must be from doseGrid (tolerance 1e-10)")
-                     o$check(! is.unsorted(object@doseGrid,
-                                           strictly=TRUE),
-                             "doseGrid must be sorted and without duplicate values")
-                     for(thisSlot in c("x", "y"))
-                         o$check(identical(object@nObs, length(slot(object, thisSlot))),
-                                 paste(thisSlot, "must have length nObs"))
-                     o$check(identical(object@nGrid, length(object@doseGrid)),
-                             "doseGrid must have length nGrid")
-                     o$check(all.equal(object@x,
-                                       object@doseGrid[object@xLevel],
-                                       tolerance=1e-10,
-                                       check.names=FALSE,
-                                       check.attributes=FALSE),
-                             "x must be doseGrid[xLevel] (tolerance 1e-10)")
+#' `Data`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' The [`Data-class`] is a class for the data input.
+#' It inherits from [`GeneralData-class`].
+#'
+#' @slot x (`numeric`)\cr the doses for the patients.
+#' @slot y (`integer`)\cr the vector of toxicity events (0 or 1 integers).
+#' @slot doseGrid (`numeric`)\cr the vector of all possible doses (sorted),
+#' i.e. the dose grid.
+#' @slot nGrid (`integer`)\cr number of gridpoints.
+#' @slot xLevel (`integer`)\cr the levels for the doses the patients have been given,
+#' w.r.t `doseGrid`.
+#' @slot placebo (`logical`)\cr if `TRUE` the first dose level
+#' in the `doseGrid`is considered as PLACEBO.
+#'
+#' @example examples/Data-class.R
+#' @export
+#' @keywords classes
```

Comment:
I think we can remove these `@keywords` tags in general. since in the `pkgdown` information we already group things nicely - this is duplicate grouping